rails-template.rb

```diff
@@ -0,0 +1,596 @@
+  doctype html
+  html lang=I18n.locale
+    = render "components/head"
+    body class=Rails.env
+      = yield
+SLIM
+
+file "app/views/components/_head.html.slim", <<~SLIM
+  head
+    meta charset="UTF-8"
+
+    / WebApp
+    meta name="viewport" content="width=device-width, initial-scale = 1.0, viewport-fit=cover"
+    / link rel="manifest" href=manifest_path
+    / link rel="apple-touch-icon" href=asset_path("app-icon.png")
+    meta name="theme-color" content="#1d2024"
+    / SEO
+    /= favicon_link_tag asset_path('favicon.svg')
+
+    / Security
+    = csrf_meta_tags
+    = csp_meta_tag
+
+    / Assets
+    = stylesheet_link_tag "application", media: "all", "data-turbo-track": "reload"
+    = javascript_include_tag "application", "data-turbo-track": "reload", defer: true
+
+    = yield :additional_head_tags if content_for? :additional_head_tags
+SLIM
+
+file "app/controllers/application_controller.rb", <<~RUBY
+  # frozen_string_literal: true
+
+  class ApplicationController < ActionController::Base
+    include Shimmer::Localizable
+    include Shimmer::RemoteNavigation
+
+    before_action :check_locale
+  end
+RUBY
+
+file "app/controllers/pages_controller.rb", <<~RUBY
+  # frozen_string_literal: true
+
+  class PagesController < ApplicationController
+    def home
+    end
+  end
+RUBY
+
+file "app/views/pages/home.html.slim", <<~SLIM
+  h1 #{app_name}
+SLIM
+
+file ".rspec", "--require spec_helper"
+
+file ""
+
+file "spec/spec_helper.rb", <<~RUBY
+  # frozen_string_literal: true
+
+  RSpec.configure do |config|
+    config.filter_run_when_matching :focus
+    config.disable_monkey_patching!
+    config.order = :random
+  end
+RUBY
+
+file "spec/rails_helper.rb", <<~RUBY
+  # frozen_string_literal: true
+
+  require "spec_helper"
+  ENV["RAILS_ENV"] ||= "test"
+  require File.expand_path("../config/environment", __dir__)
+  abort("The Rails environment is running in production mode!") if Rails.env.production?
+  require "rspec/rails"
+
+  Dir[Rails.root.join("spec/support/**/*.rb")].sort.each { |f| require f }
+
+  begin
+    ActiveRecord::Migration.maintain_test_schema!
+  rescue ActiveRecord::PendingMigrationError => e
+    puts e.to_s.strip
+    exit 1
+  end
+
+  RSpec.configure do |config|
+    config.include ActiveSupport::Testing::TimeHelpers
+    config.include ActiveJob::TestHelper
+
+    config.fixture_path = "\#{::Rails.root}/spec/fixtures"
+    config.use_transactional_fixtures = true
+    config.infer_spec_type_from_file_location!
+    config.filter_rails_from_backtrace!
+  end
+RUBY
+
+file "spec/system_helper.rb", <<~RUBY
+  # frozen_string_literal: true
+
+  require "rails_helper"
+  require "rack_session_access/capybara"
+
+  Rails.application.routes.default_url_options[:locale] = :en
+  Rails.application.routes.default_url_options[:debug] = true
+
+  RSpec.configure do |config|
+    config.before(:each, type: :system) do
+      driven_by Capybara.javascript_driver
```

Comment:
Missing alternative browser choice, for those using _Chromium_ instead of _Chrome_, unless we want to impose _Chrome_ (no strong opinion here).

```ruby
# config/environments/test.rb
  config.x.cuprite.browser = ENV["CUPRITE_BROWSER"]&.to_sym || :chrome
  config.x.cuprite.headless = ENV.fetch("HEADLESS", "true").presence_in(["1", "true"])

# spec/system_helper.rb
driven_by :cuprite, using: Rails.configuration.x.cuprite.browser

# spec/support/system/cuprite_setup.rb, in the driver's configuration
  headless: Rails.configuration.x.cuprite.headless
```